pypesto/prediction/amici_predictor.py

```diff
@@ -274,19 +292,19 @@ def _default_output(amici_outputs):
         return timepoints, outputs, outputs_sensi
 
     def _wrap_call_to_amici(self, amici_outputs, x, sensi_orders, mode,
-                            edatas):
+                            parameter_mapping, edatas):
         """
         The only purpose of this function is to encapsulate the call to amici:
         This allows to use variable scoping as a mean to clean up the memory
         after calling amici, which is beneficial if large models with large
         datasets are used.
         """
         chunk = self.amici_objective(x=x, sensi_orders=sensi_orders, mode=mode,
+                                     parameter_mapping=parameter_mapping,
                                      edatas=edatas,  return_dict=True)
         for rdata in chunk[RDATAS]:
-            amici_outputs.append({AMICI_STATUS: rdata[AMICI_STATUS],
-                                  AMICI_T: rdata[AMICI_T],
-                                  AMICI_X: rdata[AMICI_X],
-                                  AMICI_SX: rdata[AMICI_SX],
-                                  AMICI_Y: rdata[AMICI_Y],
-                                  AMICI_SY: rdata[AMICI_SY]})
+            amici_outputs.append({
+                output_field: deepcopy(rdata[output_field])
+                for output_field in self.amici_output_fields
+            })
+        del chunk
```

Comment:
Is `deepcopy` used because `chunk` remained in memory (or I guess, because after `del chunk`, maybe the value in `amici_outputs` is also deleted without `deepcopy`)? If so, is both `deepcopy` and `del` necessary? Since `chunk` should get garbage collected I guess, if there are no references to it? Just curious, fine to leave as-is

Reply:
I actually intended t go without the deepcopy and the del, and it seemed to work fine. However, in my debugger I experienced that chunk was still present i nthe memory at a later call of the function and just got overwritten... That puzzled me extremely...
Might be, that this is really only due to the debugger. However, I wanted make really, really safe that chunk is dead, dead, DEAD! :bat: No, not undead, dead. :coffin:  ... That's why deepcopy and del... :man_shrugging: